libcxx/include/__configuration/abi.h

```diff
@@ -176,6 +176,12 @@
 // ABI impact: changes the iterator type of `vector` (except `vector<bool>`).
 // #define _LIBCPP_ABI_BOUNDED_ITERATORS_IN_VECTOR
 
+// Changes the iterator type of `array` to a bounded iterator that keeps track of whether it's within the bounds of the
+// container and asserts it on every dereference and when performing iterator arithmetic.
+//
+// ABI impact: changes the iterator type of `array`.
+// #define _LIBCPP_ABI_BOUNDED_ITERATORS_IN_STD_ARRAY
```

Comment:
would it make sense to enable this setting by default under ABI v2?

By enabling this by default in ABI v2, users could control bounds checks for `std::array` iterators through `_LIBCPP_HARDENING_MODE`, without an additional ABI flag.

On the other hand, there is no negative runtime overhead of using the new bounded iterators, afaict, and the only thing that changes is the ABI name of the iterator?

Reply:
> We have to store the beginning of the array in addition to the pointer into the array itself.

Ah, I wasn't aware of that (only read the commit message, not the code)

> My preference would be to be consistent with other similar ABI settings for hardening and not enable them by default in ABI v2.

makes sense

Reply:
Updated the PR description, which was ambiguous indeed.